web-ui/view/build.ml

```diff
@@ -174,3 +184,169 @@ let ref_row ~ref_title ~short_hash ~last_updated ~status ~ref_uri ~message =
               ];
           ];
       ])
+
+let poll =
```

Comment:
I don't really understand the question. If I understand correctly this entire block of javascript needs to be wrapped in `<script>` tags. Perhaps I am not following your suggestion.

Separately, this is a short lived solution. @tmcgilchrist is helping out by wiring up ocaml-ci to use `jsoo` so that instead of great big blocks of unsafe strings, we will have a sensible way of writing javascript.

Reply:
I would suggest to move the js code to a separate file. Then, you can use [ocaml-crunch](https://github.com/mirage/ocaml-crunch/) to embed the file as a string in the OCaml binary, add a new route to serve it, and just have a `<script src="file.js">` in the html code.
You can even set up a linter / static analysis on the JS code this way, and it'll have syntactic coloration and editor support that you don't get by embedding the JS code in the OCaml code.
(I think this is what @maiste meant? — but indeed it'd be way cooler to have the JS code generated from jsoo)

Reply:
Ah ok thanks @MisterDA for the clarification -- that's a simpler and safer way to introduce this for sure and I don't think it is much extra work. 

Thanks both for your suggestions :)